drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c

```diff
@@ -465,25 +464,12 @@ static int wifi_nrf_drv_main_zep(const struct device *dev)
 		(MAX_PKT_RAM_TX_ALIGN_OVERHEAD * max_tx_aggregation);
 #endif /* CONFIG_NRF700X_DATA_TX */
 
-	status = wifi_nrf_fmac_dev_add_zep(&rpu_drv_priv_zep);
-
-	if (status != WIFI_NRF_STATUS_SUCCESS) {
-		LOG_ERR("%s: wifi_nrf_fmac_dev_add_zep failed\n",
-			__func__);
-		goto fmac_deinit;
-	}
 #ifndef CONFIG_NRF700X_RADIO_TEST
 	k_work_init_delayable(&vif_ctx_zep->scan_timeout_work,
 			      wifi_nrf_scan_timeout_work);
 #endif /* !CONFIG_NRF700X_RADIO_TEST */
 
 	return 0;
-fmac_deinit:
```

Comment:
Hi Chaitanya,
I see that most of the init functions internally invoke the free of resources in the init routines itself and removing this code makes sense. But, would it make more sense to create deinitialization routines to specifically be called during init failures rather than have each init routine clean after itself. I guess it is a coding paradigm, just a point i was pondering.

Reply:
In Zephyr the interface management is not run time i.e., there is no `de-init` in https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/net/net_if.h#L2530.